test/tagged_template_literals.coffee

```diff
@@ -1,8 +1,91 @@
 # Tagged template literals
 # ------------------------
 
-func = (text) -> "I saw: #{text}"
+assertErrorFormat = (code, expectedErrorFormat) ->
+  throws (-> CoffeeScript.run code), (err) ->
+    err.colorful = no
+    eq expectedErrorFormat, "#{err}"
+    yes
+
+func = (text, expressions...) ->
+  "text: [#{text.join ','}] expressions: [#{expressions.join ','}]"
+
+obj =
+  func: func
+
+
+test "tagged template literals: non-interpolated strings and tag function", ->
+
+  eq 'text: [single-line single quotes] expressions: []', func'single-line single quotes'
+
+  eq 'text: [single-line double quotes] expressions: []', func"single-line double quotes"
+
+  eq 'text: [single-line block string] expressions: []', func"""single-line block string"""
+
+  eq 'text: [multi-line single quotes] expressions: []', func'multi-line
+                                                               single quotes'
+
+  eq 'text: [multi-line double quotes] expressions: []', func"multi-line
+                                                               double quotes"
+
+  eq 'text: [multi-line\n block string] expressions: []', func"""
+                                                               multi-line
+                                                                block string
+                                                                 """
+
+test "tagged template literals: string prefix must be function", ->
```

Comment:
@lydell since you're around right now, are you up for a webchat about tagged template literals? I can do Skype on my github username, or Google Hangouts on ghuczynski at gmail.com

 I'm working on tagged template literals today Sunday, and I'd appreciate talking to someone with more CS compiler knowledge than me!